apollo-federation/src/sources/connect/validation/graphql/strings.rs

```diff
@@ -133,6 +133,18 @@ impl<'schema> GraphQLString<'schema> {
         Some(start..end)
     }
 
+    pub(crate) fn slice(mut self, range: Range<usize>) -> Self {
+        let data = match &mut self {
+            Self::Standard { data, .. } => data,
+            Self::Block { data, .. } => data,
+        };
+        data.raw_offset += range.start;
```

Comment:
Well that certainly would explain my incorrect locations in my tests! In another branch, where I'm using `Shape` for validations, I end up not needing `.slice()` at all. So I might refactor this to ditch that method since it's rather confusing.

Reply:
I'm just passing through offsets for now. It's not the prettiest thing, but it works, and this code is gonna change quite a bit before our next release.